Packet++/src/HttpLayer.cpp

```diff
@@ -890,30 +847,32 @@ void HttpResponseFirstLine::setVersion(HttpVersion newVersion)
 	m_Version = newVersion;
 }
 
-HttpResponseLayer::HttpResponseStatusCode HttpResponseFirstLine::parseStatusCode(const char* data, size_t dataLen)
+HttpResponseStatusCode HttpResponseFirstLine::parseStatusCode(const char* data, size_t dataLen)
 {
 	// minimum data should be 12B long: "HTTP/x.y XXX"
 	if (!data || dataLen < 12)
 	{
-		return HttpResponseLayer::HttpStatusCodeUnknown;
+		return HttpResponseStatusCode::HttpStatusCodeUnknown;
 	}
 
-	const char* statusCodeData = data + 9;
+	std::string codeString = std::string(data + 9, 3);
 
-	auto codeAsEnum = StatusCodeStringToEnumMap.find(std::string(statusCodeData, 3));
-	if (codeAsEnum == StatusCodeStringToEnumMap.end())
+	if(!codeString.empty() && std::find_if(codeString.begin(), codeString.end(), [](unsigned char c){ return !std::isdigit(c); }) == codeString.end())
```

Comment:
This should be `||`: if the string is empty **OR** it doesn't contain a digit, return `HttpResponseStatusCode::HttpStatusCodeUnknown`

Reply:
fixed.